npm/webpack-batteries-included-preprocessor/index.js

```diff
@@ -60,14 +26,21 @@ const addTypeScriptConfig = (file, options) => {
   if (!rules || !Array.isArray(rules)) return options
 
   // if we find ts-loader configured, don't add it again
-  if (hasTsLoader(rules)) return options
+  if (hasTsLoader(rules)) {
+    debug('ts-loader already configured, not adding again')
+
+    return options
+  }
 
   const TsconfigPathsPlugin = require('tsconfig-paths-webpack-plugin')
   // node will try to load a projects tsconfig.json instead of the node
-  // package using require('tsconfig'), so we alias it as 'tsconfig-aliased-for-wbip'
-  const configFile = require('tsconfig-aliased-for-wbip').findSync(path.dirname(file.filePath))
 
-  const compilerOptions = getTSCompilerOptionsForUser(configFile)
+  const getTsConfig = require('get-tsconfig')
+
+  // returns null if tsconfig cannot be found in the path/parent hierarchy
+  const configFile = getTsConfig.getTsconfig(file.filePath)
+
+  configFile ? debug(`found user tsconfig.json at ${configFile?.path} with compilerOptions: ${JSON.stringify(configFile?.config?.compilerOptions)}`) : debug('no user tsconfig.json found')
```

Comment:
we aren't doing anything with this until #31282 except feeding the webpack plugin the tsconfig file path